src/deep_learning_container.py

```diff
@@ -24,6 +24,64 @@
 
 TIMEOUT_SECS = 5
 
+TOKEN = None
+
+def requests_helper(url, headers = None, timeout = 0.1):
+    response = None
+    try:
+        if headers:
+            response = requests.get(url, headers=headers, timeout=timeout)
+        else:
+            response = requests.get(url, timeout=timeout)
+
+    except requests.exceptions.RequestException as e:
+        logging.error("Request exception: {}".format(e))
+
+    return response
+
+def requests_helper_imds(url):
+    response_text = None
+    response = None
+    headers = None
+    if TOKEN:
+        headers={"X-aws-ec2-metadata-token": TOKEN}
+    timeout = TIMEOUT_SECS
+    try:
+        while timeout >= 1:
+            if headers:
+                response = requests.get(url, headers=headers, timeout=timeout)
+            else:
+                response = requests.get(url, timeout=timeout)
+            if response:
+                break
+            timeout = timeout - 2
+
+    except requests.exceptions.RequestException as e:
+        logging.error("Request exception: {}".format(e))
+
+    if response is not None and not (400 <= response.status_code < 600):
+        response_text = response.text
+
+    return response_text
+
+def initialize_imdsv2_token():
+    response = None
+    headers = {"X-aws-ec2-metadata-token-ttl-seconds": "600"}
+    url = "http://169.254.169.254/latest/api/token"
+    timeout = TIMEOUT_SECS
+    global TOKEN
```

Comment:
nit: use of globals here will likely get confusing and potentially lead to race conditions. Why not return the token here, and pass it in as an argument to request_helper_imds?